src/commands/Utility/spotify.js

```diff
@@ -1,95 +1,136 @@
+        .addField('Playlist Name', apiResponse.name, true)
+        .addField('Created by', apiResponse.owner.display_name, true)
+        .addField('Description', apiResponse.description, true)
+        .addField('Public Playlist', ((apiResponse.public)?'Yes':'No'), true)
+        .addField('Collaborative', ((apiResponse.collaborative)?'Yes':'No'), true)
+        .addField('Followers', apiResponse.followers.total, true)
+        .addField('Track Count', apiResponse.tracks.total, true);
 
-        try {
-            data = JSON.parse(res.body);
-        } catch (e) {
-            msg.error('SpotifyAPI returned weird data. See console.');
-            bot.logger.severe(e);
-        }
+        msg.edit({content:'',embed:embed});
+        
+        if(withPreview) msg.channel.send(`Got spotify? Click below to play! ${apiResponse.external_urls.spotify}`);
 
-        if (data['type'] === 'track') {
-
-            let artists = [];
-            data['artists'].forEach(function (artist) {
-                artists.push(artist.name);
-            }, this);
-
-            let duration_m = Math.floor(data.duration_ms / 60000);
-            let duration_s = ((data.duration_ms % 60000) / 1000).toFixed(0);
-
-            let embed = bot.utils.embed(
-                'Listen now!',
-                `${msg.author.username} is currently listening to`,
-                [
-                    {
-                        name: 'Artist(s)',
-                        value: `${artists.join(', ')}`,
-                    },
-                    {
-                        name: 'Title',
-                        value: `${data.name}`,
-                    },
-                    {
-                        name: 'Explict',
-                        value: `${(data.explict) ? 'yes' : 'no'}`,
-                    },
-                    {
-                        name: 'Popularity',
-                        value: `${data.popularity} %`,
-                    },
-                    {
-                        name: 'Duration',
-                        value: (duration_s == 60 ? (duration_m + 1) + ':00' : duration_m + ':' + (duration_s < 10 ? '0' : '') + duration_s) + 'min',
-                    },
-                    {
-                        name: 'Markets',
-                        value: `Song is available on ${data.available_markets.length} markets`,
-                    },
-                ],
-                {
-                    inline: true,
-                    footer: 'Spotify Track Info',
-                    color: [30, 215, 96]
-                });
-
-            embed.setThumbnail(`${data['album']['images'][2]['url']}`);
-            embed.setURL(`${data['external_urls']['spotify']}`);
-            msg.delete();
-            msg.channel.send({ embed });
-
-        } else {
-            msg.error(`No info found for ${input}`);
-        }
+    }, function(err) {
```

Comment:
Please use lambda notation (`(...) => { ... }`) instead of function notation (`function (...) { ... }`) for the sake of consistency in code style.

Again, in this case, just `err => { ... }`

Reply:
On a second note, this isn't a requirement, but I would prefer the `promise.then(...).catch(...)` syntax to `promise.then(..., ...)`